File: frontend/src/utils/currency.ts
export type CurrencyFormatOptions = {
  locale?: string;
  currencyCode?: string;
  symbol?: string;
  minimumFractionDigits?: number;
  maximumFractionDigits?: number;
};

export function formatCurrencyAmount(
  amount: number | string | null | undefined,
  options: CurrencyFormatOptions = {}
) {
  const {
    locale = "es-AR",
    currencyCode = "ARS",
    symbol,
    minimumFractionDigits = 0,
    maximumFractionDigits = 2,
  } = options;

  const numericValue = Number(amount ?? 0);
  const safeValue = Number.isFinite(numericValue) ? numericValue : 0;

  const formatted = new Intl.NumberFormat(locale, {
    minimumFractionDigits,
    maximumFractionDigits,
  }).format(safeValue);

  if (symbol) {
    return `${symbol} ${formatted}`;
  }

  try {
    return new Intl.NumberFormat(locale, {
      style: "currency",
      currency: currencyCode,
      minimumFractionDigits,
      maximumFractionDigits,
    }).format(safeValue);
  } catch {
    return `${currencyCode} ${formatted}`;
  }
}

export function getCurrencySymbol(currencyCode?: string | null) {
  const code = String(currencyCode || "").toUpperCase();

  const symbols: Record<string, string> = {
    ARS: "$",
    USD: "US$",
    EUR: "€",
    CLP: "$",
    MXN: "$",
  };

  return symbols[code] || "$";
}

File: backend/app/core/currency.py
from decimal import Decimal
from typing import Any


DEFAULT_SYMBOLS = {
    "ARS": "$",
    "USD": "US$",
    "EUR": "€",
    "CLP": "$",
    "MXN": "$",
}


def get_currency_symbol(currency_code: str | None, fallback: str = "$") -> str:
    code = str(currency_code or "").upper().strip()
    return DEFAULT_SYMBOLS.get(code, fallback)


def format_money(
    value: Any,
    currency_code: str | None = "USD",
    symbol: str | None = None,
) -> str:
    try:
        amount = Decimal(str(value or 0))
    except Exception:
        amount = Decimal("0")

    code = str(currency_code or "USD").upper()
    final_symbol = symbol or get_currency_symbol(code)

    formatted = f"{amount:,.2f}"
    formatted = formatted.replace(",", "X").replace(".", ",").replace("X", ".")

    return f"{final_symbol} {formatted}"
